pandas/core/frame.py

```diff
@@ -374,8 +376,12 @@ def _constructor_expanddim(self):
     # ----------------------------------------------------------------------
     # Constructors
 
-    def __init__(self, data=None, index=None, columns=None, dtype=None,
-                 copy=False):
+    def __init__(self,
+                 data=None,
+                 index: ArrayLike = None,
+                 columns: ArrayLike = None,
+                 dtype: Dtype = None,
```

Comment:
TODO:
- Figure out proper type for index and columns
- Figure out proper type for data